src/AudioToolbox/MusicSequence.cs

```diff
@@ -274,6 +281,33 @@ public double GetBeatsForSeconds (double seconds)
 				return beats;
 			return 0;
 		}
+
+		[DllImport (Constants.AudioToolboxLibrary)]
+		extern static /* OSStatus */ MusicPlayerStatus MusicSequenceSetUserCallback (/* MusicSequence */ IntPtr inSequence, MusicSequenceUserCallbackProxy inCallback, /* void * */ IntPtr inClientData);
+
+		MusicSequenceUserCallback userCallback;
+
+		public void SetUserCallback (MusicSequenceUserCallback callback)
+		{
+			if (MusicSequenceSetUserCallback (handle, UserCallbackProxy, IntPtr.Zero) == MusicPlayerStatus.Success)
```

Comment:
You're not storing the `UserCallbackProxy` delegate somewhere, so the GC will free it.

You need to create a static `UserCallbackProxy` field, store the delegate there, and use that in the P/Invoke.